runtime/libffi/closures.c

```diff
@@ -492,8 +492,8 @@ dlmmap_locked (void *start, size_t length, int prot, int flags, off_t offset)
 	  close (execfd);
 	  goto retry_open;
 	}
-      ftruncate (execfd, offset);
-      return MFAIL;
+      if (ftruncate (execfd, offset))
+        return MFAIL;
```

Comment:
The old code unconditionally returned `MFAIL` - it must continue to do so:
```c
      if (ftruncate (execfd, offset))
        return MFAIL;
      return MFAIL;
```
Similarly, the change to lines 507-508 must unconditionally return `start`:
```c
      if (ftruncate (execfd, offset));
        return start;
      return start;
```
Normally the `if` statements would explicitly compare with zero, but following the pattern (from upstream) on line 480 is reasonable.